src/modules/Builder/features/ActivityItems/ConditionalPanel/ConditionalPanel.tsx

```diff
@@ -0,0 +1,108 @@
+import { useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { useTranslation, Trans } from 'react-i18next';
+import { Collapse } from '@mui/material';
+
+import {
+  ConditionalLogic,
+  OptionCondition,
+  RangeValueCondition,
+  SingleAndMultipleSelectOption,
+  SingleValueCondition,
+} from 'shared/state';
+import { getEntityKey } from 'shared/utils';
+import { Svg } from 'shared/components';
+import {
+  StyledBodyLarge,
+  StyledClearedButton,
+  StyledFlexColumn,
+  StyledFlexTopCenter,
+  theme,
+} from 'shared/styles';
+import {
+  ItemResponseType,
+  CONDITION_TYPES_TO_HAVE_RANGE_VALUE,
+  CONDITION_TYPES_TO_HAVE_SINGLE_VALUE,
+} from 'shared/consts';
+import { useCurrentActivity } from 'modules/Builder/pages/BuilderApplet/BuilderApplet.hooks';
+import { ItemFormValues } from 'modules/Builder';
+
+export const ConditionalPanel = ({ condition }: { condition?: ConditionalLogic }) => {
+  const { t } = useTranslation('app');
+  const [isExpanded, setExpanded] = useState(true);
+  const { watch } = useFormContext();
+  const { fieldName } = useCurrentActivity();
+
+  const items = watch(`${fieldName}.items`);
+  const currentItem = items?.find(
+    (item: ItemFormValues) => getEntityKey(item) === condition?.itemKey,
+  );
+
+  return (
+    <Collapse in={isExpanded} timeout={0} collapsedSize="4.8rem">
+      <StyledFlexTopCenter sx={{ gap: '1rem' }}>
+        <StyledClearedButton
+          sx={{ p: theme.spacing(1) }}
+          onClick={() => setExpanded((prevExpanded) => !prevExpanded)}
+        >
+          <Svg id={isExpanded ? 'navigate-up' : 'navigate-down'} />
+        </StyledClearedButton>
+        <StyledBodyLarge>
+          <Trans i18nKey="conditionalLogicPanelTitle">
+            If
+            <strong>
+              {' '}
+              <>{{ match: condition?.match }}</>{' '}
+            </strong>
+            of the “if” rules below are
+            <strong>
+              {' '}
+              true, show item: <>{{ name: currentItem.name }}</>
+            </strong>
+          </Trans>
+        </StyledBodyLarge>
+      </StyledFlexTopCenter>
+      {isExpanded && (
+        <StyledFlexColumn sx={{ mt: theme.spacing(1.2) }}>
+          {condition?.conditions?.map(({ key, type, itemName, payload }) => {
+            const relatedItem = items?.find(
```

Comment:
I would create object before map and instead of find get by obj[itemName] and for currentItem  as well